src/Microsoft.OData.Core/RawValueWriter.cs

```diff
@@ -173,8 +173,8 @@ private void InitializeTextWriter()
             {
                 nonDisposingStream = MessageStreamWrapper.CreateNonDisposingStream(this.stream);
             }
-
-            this.textWriter = new StreamWriter(nonDisposingStream, this.encoding);
+            
+            this.textWriter = new StreamWriter(nonDisposingStream, this.encoding) { NewLine = this.settings.MultipartNewLine };
             this.jsonWriter = new JsonWriter(this.textWriter, isIeee754Compatible: false);

```

Comment:
I'd like to see the test cases to cover this change. Maybe we can use "|" as a new Line to test the result.

Reply:
Correct me if I am wrong but that would only test that System.IO.TextWriter is indeed honoring the NewLine setting. 
To test that the setting flows from `new ODataBatchWriter ` into the TextWriter would require to be able to mock a TextWriter which is to my understand way out of the scope of this.